### PR TITLE
Display task health in the Web UI.

### PR DESCRIPTION
**Pull request by janisz**
Fixes: [MESOS-7961](https://issues.apache.org/jira/browse/MESOS-7961)

screencapture-localhost-5050-1505330708402

### CODE REVIEW

**Comment by lilysamimi** [comment]
Would be interested in talking with the folks that created this UI component. I'm giving out $10 Amazon gift cards to individuals who participate in a user experience research study with me pertaining to the Mesos UI. Please DM me. I'm at MesosCon today and tomorrow and around the Mesosphere booth.

**Comment by bmahler** [comment]
Thanks janisz, this looks pretty good. However, this needs to be added to several places:

- The global tasks table (you covered this)
- The framework-specific task table (when clicking into a framework)
- The executor-specific task table within an agent (when clicking into an agent, then a framework, then an executor)

I'm also a bit curious about whether it's a good idea to re-use the health string as a class, since it seems easy to break vs only specifying the display string in the html page.

Will you be able to continue working on this?

**Comment by janisz** [comment]
@bmahler Thanks for the review. Most of the issues are minor and easy to fix. I do it shortly.

**Comment by Vish0007** [comment]
@janisz  the UI looks good and health check does add value.
would you describe how the "Task Name" and "Health" are different, i can see health status in both places.

**Comment by janisz** [comment]
@Vish0007 To create this screenshot I created 3 applications called `unhealthy`, `healthy` and `unknown` so it's just a coincidence that the name has the same value as tasks health. When I fix all issues I'll prepare gif to present healthy is available in all places and I'll use more generic names in order to avoid confusion.

**Comment by Vish0007** [comment]
@janisz sounds great.

File: src/webui/master/static/home.html [hunk 1]
```diff
@@ -162,14 +162,15 @@
           <th data-key="name">Task Name</th>
```
Comment (Vishant0031) [inline src/webui/master/static/home.html]: is this task id (MESOS_TASK_ID?), should this be "Task Id"?

Comment (janisz) [reply src/webui/master/static/home.html]: Task ID is one line above

**Comment by janisz** [comment]
* [x] The global tasks table
* [x] The framework-specific task table (when clicking into a framework)
* [x] The executor-specific task table within an agent (when clicking into an agent, then a framework, then an executor)

> I'm also a bit curious about whether it's a good idea to re-use the health string as a class, since it seems easy to break vs only specifying the display string in the html page.

I removed CSS. I'm not sure if red/green looks good in Mesos blue palette. @lilysamimi Any idea how to present health state?


health



<details>

I used Marathon to deploy 3 applications with 3 types of health state.

```
 curl -X PUT -d@health_groups.json localhost:8080/v2/groups/
```

```json
{
   "apps":[
      {
         "id":"/apache",
         "cmd":"sleep 1000",
         "cpus":1,
         "mem":128,
         "disk":0,
         "instances":1,
         "healthChecks":[
            {
               "gracePeriodSeconds":1,
               "intervalSeconds":1,
               "maxConsecutiveFailures":300,
               "command":{
                  "value":"true"
               },
               "protocol":"COMMAND"
            }
         ]
      },
      {
         "id":"/tomcat",
         "cmd":"sleep 1000",
         "cpus":1,
         "mem":128,
         "disk":0,
         "instances":1,
         "healthChecks":[
            {
               "gracePeriodSeconds":1,
               "intervalSeconds":1,
               "maxConsecutiveFailures":300,
               "command":{
                  "value":"false"
               },
               "protocol":"COMMAND"
            }
         ]
      },
      {
         "id":"/mysql",
         "cmd":"sleep 1000",
         "cpus":1,
         "mem":128,
         "disk":0,
         "instances":1
      }
   ]
}
```

</details>

**Comment by bmahler** [comment]
@janisz I noticed you updated the patch, is it ready for another review? Would be great to keep the colors as you had before. I was just asking about the approach used :)

Also can you add a gif or video of your screen to show the testing of it?

**Comment by janisz** [comment]
@bmahler I added some colours. 

health

**Comment by janisz** [comment]
@kwiatkk1 Thanks for the review. I applied your suggestions.
@bmahler Changed `Unknown` to `-`


out-2

File: src/webui/master/static/js/controllers.js [hunk 2]
```diff
@@ -838,6 +839,15 @@
             });
           }
 
+          function setHealth(tasks) {
+            _.each(tasks, function(task) {
+              var lastStatus = _.last(task.statuses);
+              if (lastStatus) {
+                  task.healthy = lastStatus.healthy;
```
Comment (bmahler) [inline src/webui/master/static/js/controllers.js]: 2 space indents

File: src/webui/master/static/js/app.js [hunk 3]
```diff
@@ -88,6 +88,14 @@
         return state.substring(5);
       };
     })
+    .filter('taskHealth', function() {
+      return function(health) {
+        if (health != null) {
+           return health ? "healthy" : "unhealthy";
```
Comment (bmahler) [inline src/webui/master/static/js/app.js]: 2 space indent, would also be good to warn the reader here that this value is relied on for the CSS

**Pull request closed without merging**